packages/material-ui/src/Slider/Slider.js

```diff
@@ -1,1035 +1,312 @@
-  /* Pseudo-class applied to the root and thumb element if `disabled={true}`. */
-  disabled: {},
-  /* Styles applied to the rail element. */
-  rail: {
-    display: 'block',
+    ...(props.marked && {
+      marginBottom: 20,
+      ...(props.orientation === 'vertical' && {
+        marginBottom: 'auto',
+        marginRight: 20,
+      }),
+    }),
+  };
+});
+
+export const SliderRail = styled('span', { shouldForwardProp })((props) => ({
+  display: 'block',
+  position: 'absolute',
+  width: '100%',
+  height: 2,
+  borderRadius: 1,
+  backgroundColor: 'currentColor',
+  opacity: 0.38,
+  ...(props.orientation === 'vertical' && {
+    height: '100%',
+    width: 2,
+  }),
+  ...(props.track === 'inverted' && {
+    opacity: 1,
+  }),
+}));
+
+export const SliderTrack = styled('span', { shouldForwardProp })((props) => ({
+  display: 'block',
+  position: 'absolute',
+  height: 2,
+  borderRadius: 1,
+  backgroundColor: 'currentColor',
+  ...(props.orientation === 'vertical' && {
+    width: 2,
+  }),
+  ...(props.track === false && {
+    display: 'none',
+  }),
+  ...(props.track === 'inverted' && {
+    backgroundColor:
+      // Same logic as the LinearProgress track color
+      props.theme.palette.type === 'light'
+        ? lighten(props.theme.palette.primary.main, 0.62)
+        : darken(props.theme.palette.primary.main, 0.5),
+  }),
+}));
+
+export const SliderThumb = styled('span', { shouldForwardProp })((props) => ({
+  position: 'absolute',
+  width: 12,
+  height: 12,
+  marginLeft: -6,
+  marginTop: -5,
+  boxSizing: 'border-box',
+  borderRadius: '50%',
+  outline: 0,
+  backgroundColor: 'currentColor',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  transition: props.theme.transitions.create(['box-shadow'], {
+    duration: props.theme.transitions.duration.shortest,
+  }),
+  '::after': {
     position: 'absolute',
-    width: '100%',
-    height: 2,
-    borderRadius: 1,
-    backgroundColor: 'currentColor',
-    opacity: 0.38,
-    '$vertical &': {
-      height: '100%',
-      width: 2,
-    },
+    content: '""',
+    borderRadius: '50%',
+    // reach 42px hit target (2 * 15 + thumb diameter)
+    left: -15,
+    top: -15,
+    right: -15,
+    bottom: -15,
   },
-  /* Styles applied to the track element. */
-  track: {
-    display: 'block',
-    position: 'absolute',
-    height: 2,
-    borderRadius: 1,
-    backgroundColor: 'currentColor',
-    '$vertical &': {
-      width: 2,
+  ':hover': {
+    boxShadow: `0px 0px 0px 8px ${fade(props.theme.palette.primary.main, 0.16)}`,
+    '@media (hover: none)': {
+      boxShadow: 'none',
     },
   },
-  /* Styles applied to the track element if `track={false}`. */
-  trackFalse: {
-    '& $track': {
-      display: 'none',
+  ...(props.focusVisible && {
+    boxShadow: `0px 0px 0px 8px ${fade(props.theme.palette.primary.main, 0.16)}`,
+    '@media (hover: none)': {
+      boxShadow: 'none',
     },
+  }),
+  '&.MuiSlider--active': {
```

Comment:
To avoid a breaking change?
```suggestion
  '&.Mui-active': {
```